packages/core/src/browser/quick-open/quick-open-action.ts

```diff
@@ -0,0 +1,116 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Disposable } from '../../common/disposable';
+import { injectable, unmanaged } from 'inversify';
+
+export interface QuickOpenAction extends Disposable {
+    id: string;
+    label: string;
+    tooltip: string;
+    class: string | undefined;
+    enabled: boolean;
+    checked: boolean;
+    radio: boolean;
+    // tslint:disable-next-line:no-any
```

Comment:
`run(event?: { item?: QuickOpenItem }): PromiseLike<any>;` would it be correct type?

Reply:
Done: I added `MonacoQuickOpenAction` to provide ability to use `QuickOpenItem` here